cmd/generator/main.go

```diff
@@ -253,14 +273,24 @@ func getLatestRelease(ctx context.Context, client *github.Client, repoName strin
 	return latestRelease, nil
 }
 
-func getIcon(ctx context.Context, url string) ([]byte, error) {
-	logger.Debugf("fetching icon url %s", url)
+func getIcon(ctx context.Context, icon string) ([]byte, error) {
+	if strings.HasPrefix(icon, "http") {
+		logger.Debugf("fetching icon from url %s", icon)
+
+		resp, err := http.Get(icon)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to download plugin icon at %s", icon)
+		}
+		defer resp.Body.Close()
+
+		return ioutil.ReadAll(resp.Body)
```

Comment:
Not sure if it makes any difference but should this return `nil` for err: 

```suggestion
		return ioutil.ReadAll(resp.Body), nil
```

Reply:
`ioutil.ReadAll` is [defined](https://golang.org/pkg/io/ioutil/#ReadAll) as the following, so I'm effectively returning any error from reading the body:
```go
func ReadAll(r io.Reader) ([]byte, error)
```